tests/python_package_test/test_dask.py

```diff
@@ -0,0 +1,211 @@
+
+import lightgbm
+import lightgbm.dask as dlgbm
+
+data_output = ['array', 'scipy_csr_matrix', 'dataframe']
+data_centers = [[[-4, -4], [4, 4]], [[-4, -4], [4, 4], [-4, 4]]]
+
+pytestmark = [
+    pytest.mark.skipif(os.getenv("TASK", "") == "mpi", reason="Fails to run with MPI interface")
+]
+
+
+@pytest.fixture()
+def listen_port():
+    listen_port.port += 10
+    return listen_port.port
+
+
+listen_port.port = 13000
+
+
+def _create_data(objective, n_samples=100, centers=2, output='array', chunk_size=50):
+    if objective == 'classification':
+        X, y = make_blobs(n_samples=n_samples, centers=centers, random_state=42)
+    elif objective == 'regression':
+        X, y = make_regression(n_samples=n_samples, random_state=42)
+    else:
+        raise ValueError(objective)
+    rnd = np.random.RandomState(42)
+    w = rnd.random(X.shape[0]) * 0.01
+
+    if output == 'array':
+        dX = da.from_array(X, (chunk_size, X.shape[1]))
+        dy = da.from_array(y, chunk_size)
+        dw = da.from_array(w, chunk_size)
+    elif output == 'dataframe':
+        X_df = pd.DataFrame(X, columns=['feature_%d' % i for i in range(X.shape[1])])
+        y_df = pd.Series(y, name='target')
+        dX = dd.from_pandas(X_df, chunksize=chunk_size)
+        dy = dd.from_pandas(y_df, chunksize=chunk_size)
+        dw = dd.from_array(w, chunksize=chunk_size)
+    elif output == 'scipy_csr_matrix':
+        dX = da.from_array(X, chunks=(chunk_size, X.shape[1])).map_blocks(scipy.sparse.csr_matrix)
+        dy = da.from_array(y, chunks=chunk_size)
+        dw = da.from_array(w, chunk_size)
+    else:
+        raise ValueError("Unknown output type %s" % output)
+
+    return X, y, w, dX, dy, dw
+
+
+@pytest.mark.parametrize('output', data_output)
+@pytest.mark.parametrize('centers', data_centers)
+def test_classifier(output, centers, client, listen_port):  # noqa
+    X, y, w, dX, dy, dw = _create_data('classification', output=output, centers=centers)
+
+    a = dlgbm.LGBMClassifier(time_out=5, local_listen_port=listen_port)
+    a = a.fit(dX, dy, sample_weight=dw, client=client)
+    p1 = a.predict(dX)
+    s1 = accuracy_score(dy, p1)
+    p1 = p1.compute()
+
+    b = lightgbm.LGBMClassifier()
+    b.fit(X, y, sample_weight=w)
+    p2 = b.predict(X)
+    s2 = b.score(X, y)
+
+    assert_eq(s1, s2)
+
+    assert_eq(p1, p2)
+    assert_eq(y, p1)
+    assert_eq(y, p2)
+
+
+@pytest.mark.parametrize('output', data_output)
+@pytest.mark.parametrize('centers', data_centers)
+def test_classifier_proba(output, centers, client, listen_port):  # noqa
+    X, y, w, dX, dy, dw = _create_data('classification', output=output, centers=centers)
+
+    a = dlgbm.LGBMClassifier(time_out=5, local_listen_port=listen_port)
+    a = a.fit(dX, dy, sample_weight=dw, client=client)
+    p1 = a.predict_proba(dX)
+    p1 = p1.compute()
+
+    b = lightgbm.LGBMClassifier()
+    b.fit(X, y, sample_weight=w)
+    p2 = b.predict_proba(X)
+
+    assert_eq(p1, p2, atol=0.3)
```

Comment:
Is it possible to increase precision?

Reply:
I think we can resolve this after this PR